src/mame/netronics/exp85.cpp

```diff
@@ -67,50 +84,67 @@ class exp85_state : public driver_device
 	DECLARE_INPUT_CHANGED_MEMBER( trigger_rst75 );
 
 private:
-	required_device<i8085a_cpu_device> m_maincpu;
-	required_device<rs232_port_device> m_rs232;
-	required_device<cassette_image_device> m_cassette;
-	required_device<speaker_sound_device> m_speaker;
-	required_memory_region m_rom;
-
 	virtual void machine_start() override;
 
 	uint8_t i8355_a_r();
 	void i8355_a_w(uint8_t data);
 	int sid_r();
 	void sod_w(int state);
 
-	/* cassette state */
-	bool m_tape_control;
+	void status_out(u8 status);
+
 	void exp85_io(address_map &map);
 	void exp85_mem(address_map &map);
+
+	void to_change(int to);
+
+	TIMER_CALLBACK_MEMBER(trap_delay);
+
+	// Member variables
+	required_device<i8085a_cpu_device> m_maincpu;
+	required_device<i8355_device> m_i8355;
+	required_device<i8155_device> m_i8155;	
+	required_device<rs232_port_device> m_rs232;
+	required_device<cassette_image_device> m_cassette;
+	required_device<speaker_sound_device> m_speaker;
+	required_memory_region m_rom;
+	required_device<ram_device> m_extended_ram;
+
+	bool m_is_preparing_interrupt_call;
+	bool m_ignore_timer_out;
+	bool m_tape_control;
+    emu_timer *m_timer;
 };
 
-/* Memory Maps */
 
+/* Memory Maps */
 void exp85_state::exp85_mem(address_map &map)
 {
 	map.unmap_value_high();
-	map(0x0000, 0x07ff).bankr("bank1");
+	// Extended RAM or mapped monitor ROM (only during interrupt and reset)
+	map(0x0000, 0x2000).bankrw("bank1");
```

Comment:
The problem with using `bankrw` here is that it will allow writing to the ROM, which shouldn’t be possible.  You should really use a memory view and either install the ROM in one entry and the RAM in another, or install one directly in the address space and the other in a memory view entry.

You can see a relatively simple use of a memory view in the Jaleco Psychic 5 driver (find the uses of `m_vrambank` in the file):
https://github.com/mamedev/mame/blob/mame0261/src/mame/jaleco/psychic5.cpp

You can find other uses in various places by searching for `memory_view`.

Reply:
Thanks! I have changed the bank for a memory view and I'm using it to switch between the RAM and the ROM mirror.